components/widgets/inputs/dropdown_input_hybrid.tsx

```diff
@@ -115,74 +115,60 @@ const DropdownInputHybrid = <T extends ValueType>(props: Props<T>) => {
 
     useEffect(() => {
         if (!inputValue && !focused && !inputFocused) {
-            onDropdownChange(props.defaultValue);
+            onDropdownChange(defaultValue);
             showTextInput('');
         }
     }, [focused, inputFocused]);
 
-    const getMenuStyles = () => {
-        if (showInput) {
-            return {
+    const getMenuStyles = () =>
+        showInput ?
+            {
                 menu: (provided: CSSProperties) => ({
                     ...provided,
                     width: containerRef.current ? `${containerRef.current.offsetWidth}px` : '0px',
                     left: inputRef.current ? `-${inputRef.current.offsetWidth}px` : '0px',
                 }),
-            };
-        }
-        return {};
-    }
+            }
+            : {};
 
     const onInputBlur = () => setInputFocused(false);
 
     const onInputFocus = () => setInputFocused(true);;
 
     const onDropdownInputFocus = (event: React.FocusEvent<HTMLElement>) => {
-        const {onFocus} = props;
-        
         setFocused(true);
 
-        if (onFocus) {
-            onFocus(event);
-        }
+        props.onFocus?.(event);
     };
 
     const onDropdownInputBlur = (event: React.FocusEvent<HTMLElement>) => {
-        const {onBlur} = props;
-
         setFocused(false);
 
-        if (onBlur) {
-            onBlur(event);
-        }
+        props.onBlur?.(event);
     };
 
-    const onValueChange = async (event: T) => {
-        const {onDropdownChange} = props;
-
+    const onValueChange = (event: T) => {
         showTextInput(event.value);
 
-        if (onDropdownChange) {
-            onDropdownChange(event);
-        }
+        props.onDropdownChange(event)
```

Comment:
this is already destructured from `props` above, so you can just call it directly

```suggestion
        onDropdownChange(event)
```